addons/mail/static/src/messaging/entity/composer/composer.js

```diff
@@ -203,23 +285,32 @@ function ComposerFactory({ Entity }) {
         }
 
         /**
+         * @param {string} mentionKeyword
          * @private
-         * @param {string} content html content
-         * @returns {integer[]} list of mentioned partner Ids (not duplicate)
          */
-        _getMentionedPartnerIdsFromHtml(content) {
-            const parser = new window.DOMParser();
-            const node = parser.parseFromString(content, 'text/html');
-            const mentions = [...node.querySelectorAll('.o_mention')];
-            const allPartnerIds = mentions
-                .filter(mention =>
-                    (
-                        mention.dataset.oeModel === 'res.partner' &&
-                        !isNaN(Number(mention.dataset.oeId))
-                    )
-                )
-                .map(mention => Number(mention.dataset.oeId));
-            return [...new Set(allPartnerIds)];
+        async _getMentionSuggestions(mentionKeyword) {
+            const mentions = await this.env.rpc({
+                model: 'res.partner',
+                method: 'get_mention_suggestions',
+                kwargs: {
+                    limit: 3,
+                    search: mentionKeyword,
+                },
+            }, { shadow: true });
+
+            this.update({
+                activeId: 0,
+                mentionSuggestions: [['unlink-all']],
+            });
+            for (const mention of mentions[0]) {
+                const { email, id, name } = mention;
+                const mentionSuggestion = this.env.entities.Partner.insert({ email, id, name });
+                this.update({ mentionSuggestions: [['link', mentionSuggestion]] })
+            }
+
+            if (this.mentionSuggestions.length > 0) {
+                this.update({ activeId: this.mentionSuggestions[0].id });
+            }
```

Comment:
The update at line 301 could be combined with the one at 312 to only call it once (better to "batch" stuff as much as possible to optimize compute and related methods).

Similarly, the one at line 308 could be merged with the one at 312 too.

So for `mentionSuggestions`, instead of "unlink-all" and then "link" in loop -> do only one "insert-and-replace" at the end, by giving it the complete list of partner data. This is usually done with a "map", see for example the use of this command for "attachments" in `message.js`.